### PR TITLE
update abnf

### PR DESCRIPTION
**Pull request by chrisspre**
- simplify operation syntax
- rename qualifiedName to qualifiedIdentifier
- add abstract type modifier and inheritance
- add string and decimal facet
- fix annotation value casing

### CODE REVIEW

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 1]
```diff
@@ -111,25 +120,27 @@ propertyName     = identifier / DQUOTE 1*CHAR DQUOTE
...
+identInitial        = ALPHA / "_" ; Note: actually all Unicode letters
 
-identSubsequent = identInitial / DIGIT
+identSubsequent     = identInitial / DIGIT
 
-number          = DIGIT *DIGIT ["." *DIGIT ]
+number              = DIGIT *DIGIT ["." *DIGIT ]
 
-ALPHA  = %x41-5A / %x61-7A
+integer             = DIGIT *DIGIT
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:133]: "integer" should allow negative values, and for its use in `Scale` this is actually needed.

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Note: to increase readability of the grammar, whitespace is not reflected
 ```ABNF
 model        = [ namespace ] *include *modelElement
 
-namespace    = %s"namespace" qualifiedName
+namespace    = %s"namespace" qualifiedIdentifier
```
[the remark this answers is not in the document]
Comment (chrisspre) [reply docs/rsdl/rapid-pro-rsdl-abnf.md:34]: ok. will do. 

And would you also want to change the production called 'identifier" to "name" or leave it as is and 
have identifier + qualifiedName ?

[the remark this answers is not in the document]
Comment (ralfhandl) [reply docs/rsdl/rapid-pro-rsdl-abnf.md]: I'm fine with "identifier", it's official name in CSDL is [Simple Identifier](https://docs.oasis-open.org/odata/odata-csdl-json/v4.01/odata-csdl-json-v4.01.html#sec_SimpleIdentifier)

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 3]
```diff
@@ -111,25 +120,27 @@ propertyName     = identifier / DQUOTE 1*CHAR DQUOTE
...
+identifier          = identInitial *identSubsequent
 
-identInitial    = ALPHA / "_" ; Note: actually all Unicode letters
+identInitial        = ALPHA / "_" ; Note: actually all Unicode letters
 
-identSubsequent = identInitial / DIGIT
+identSubsequent     = identInitial / DIGIT
 
-number          = DIGIT *DIGIT ["." *DIGIT ]
+number              = DIGIT *DIGIT ["." *DIGIT ]
```
[the remark this answers is not in the document]
Comment (ralfhandl) [reply docs/rsdl/rapid-pro-rsdl-abnf.md]: I'd allow exponential notation, we are borrowing from JSON anyway for annotation values, and JSON numbers allow exponential notation.

Opened https://github.com/oasis-open/odata-rapid/issues/228

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ modelElement = structuredType / enumType / service
 ### Structured Type
 
 ```ABNF
-structuredType       = annotations %s"type" identifier "{" *structuredTypeMember "}"
+structuredType       = annotations [ %s"abstract" ]  %s"type" identifier [ %"extends" qualifiedName       ] "{" *structuredTypeMember "}"
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:44]:
```suggestion
structuredType       = annotations [ %s"abstract" ]  %s"type" identifier [ %"extends" qualifiedName ] "{" *structuredTypeMember "}"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 5]
```diff
@@ -79,11 +87,11 @@ service              = %s"service" "{" 1*serviceMember "}"
 
 serviceMember        = entitySet / singleton / serviceOperation
 
-entitySet            = identifier ":" "[" qualifiedName "]"
+entitySet            = identifier ":" "[" qualifiedName       "]"
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:90]:
```suggestion
entitySet            = identifier ":" "[" qualifiedName "]"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md (hunk 4 above)
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md]:
```suggestion
structuredType       = annotations [ %s"abstract" ]  %s"type" identifier [ %"extends" qualifiedName ] "{" *structuredTypeMember "}"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 6]
```diff
@@ -31,7 +31,7 @@ Note: to increase readability of the grammar, whitespace is not reflected
 ```ABNF
 model        = [ namespace ] *include *modelElement
 
-namespace    = %s"namespace" qualifiedName
+namespace    = %s"namespace" qualifiedName
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:34]:
```suggestion
namespace    = %s"namespace" qualifiedName
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 7]
```diff
@@ -41,7 +41,7 @@ modelElement = structuredType / enumType / service
 ### Structured Type
 
 ```ABNF
-structuredType       = annotations %s"type" identifier "{" *structuredTypeMember "}"
+structuredType       = annotations [ %s"abstract" ]  %s"type" identifier [ %"extends" qualifiedName ] "{" *structuredTypeMember "}"
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:44]:
```suggestion
structuredType       = annotations [ %s"abstract" ] %s"type" identifier [ %"extends" qualifiedName ] "{" *structuredTypeMember "}"
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 8]
```diff
@@ -51,15 +51,23 @@ propertyModifier     = %s"key"
 
 typeReference        = typeName [ "?" ] / "[" typeName [ "?" ] "]"
 
-typeName             = builtInType / %s"Edm" "." identifier / qualifiedName
+typeName             = builtInType / %s"Edm" "." identifier / qualifiedName
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:54]:
```suggestion
typeName             = builtInType / %s"Edm" "." identifier / qualifiedName
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 9]
```diff
@@ -81,9 +89,9 @@ serviceMember        = entitySet / singleton / serviceOperation
 
 entitySet            = identifier ":" "[" qualifiedName "]"
 
-singleton            = identifier ":" qualifiedName
+singleton            = identifier ":" qualifiedName
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:92]:
```suggestion
singleton            = identifier ":" qualifiedName
```

**Review by ralfhandl — APPROVED** [review]

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 10]
```diff
@@ -111,25 +120,27 @@ propertyName     = identifier / DQUOTE 1*CHAR DQUOTE
...
+identInitial        = ALPHA / "_" ; Note: actually all Unicode letters
 
-identInitial    = ALPHA / "_" ; Note: actually all Unicode letters
+identSubsequent     = identInitial / DIGIT
 
-identSubsequent = identInitial / DIGIT
+number              = [ "-" ] DIGIT *DIGIT ["." *DIGIT ]
 
-number          = DIGIT *DIGIT ["." *DIGIT ]
+integer             = [ "-" ] DIGIT *DIGIT
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:133]:
```suggestion
integer             = [ "+" / "-" ] DIGIT *DIGIT
```

File: docs/rsdl/rapid-pro-rsdl-abnf.md [hunk 11]
```diff
@@ -111,25 +120,27 @@ propertyName     = identifier / DQUOTE 1*CHAR DQUOTE
...
+identInitial        = ALPHA / "_" ; Note: actually all Unicode letters
 
-identInitial    = ALPHA / "_" ; Note: actually all Unicode letters
+identSubsequent     = identInitial / DIGIT
 
-identSubsequent = identInitial / DIGIT
+number              = [ "-" ] DIGIT *DIGIT ["." *DIGIT ]
 
-number          = DIGIT *DIGIT ["." *DIGIT ]
+integer             = [ "+" / "-" ] DIGIT *DIGIT
```
Comment (ralfhandl) [inline docs/rsdl/rapid-pro-rsdl-abnf.md:133]:
```suggestion
integer             = [ "-" ] DIGIT *DIGIT
```

**Pull request merged**